### ISSUE TITLE
NPE on startup

### ISSUE THREAD

**Issue by MattPumphrey** [linkedin/cruise-control#760]
[2019-06-11 14:31:46,579] INFO Kafka version : 2.0.0 (org.apache.kafka.common.utils.AppInfoParser)
[2019-06-11 14:31:46,579] INFO Kafka commitId : 3402a8361b734732 (org.apache.kafka.common.utils.AppInfoParser)
[2019-06-11 14:31:46,757] INFO Cluster ID: K48tg2h1ShW40q8-99TgvQ (org.apache.kafka.clients.Metadata)
Exception in thread "main" java.lang.NullPointerException
	at com.linkedin.kafka.cruisecontrol.monitor.sampling.CruiseControlMetricsReporterSampler.refreshPartitionAssignment(CruiseControlMetricsReporterSampler.java:186)
	at com.linkedin.kafka.cruisecontrol.monitor.sampling.CruiseControlMetricsReporterSampler.configure(CruiseControlMetricsReporterSampler.java:253)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.getConfiguredInstance(KafkaCruiseControlConfig.java:1413)
	at com.linkedin.kafka.cruisecontrol.monitor.sampling.MetricFetcherManager.<init>(MetricFetcherManager.java:129)
	at com.linkedin.kafka.cruisecontrol.monitor.sampling.MetricFetcherManager.<init>(MetricFetcherManager.java:69)
	at com.linkedin.kafka.cruisecontrol.monitor.task.LoadMonitorTaskRunner.<init>(LoadMonitorTaskRunner.java:74)
	at com.linkedin.kafka.cruisecontrol.monitor.LoadMonitor.<init>(LoadMonitor.java:177)
	at com.linkedin.kafka.cruisecontrol.monitor.LoadMonitor.<init>(LoadMonitor.java:127)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControl.<init>(KafkaCruiseControl.java:124)
	at com.linkedin.kafka.cruisecontrol.async.AsyncKafkaCruiseControl.<init>(AsyncKafkaCruiseControl.java:75)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControlMain.main(KafkaCruiseControlMain.java:71)

I am using 2.0.50 and still experiencing these issues, it seems to have been noticed in previous versions per #731.  We would like to use this, but I have been only able to get it to work when I install it on the brokers themselves and not when using as a stand alone.

**Comment by ninthnails** [comment]
Same problem here, NPE at start up with no clear indication what is causing it. Using 2.0.50 on OpenJDK 11. 
```
Exception in thread "main" java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1316)
	at java.base/java.util.Properties.setProperty(Properties.java:225)
	at com.linkedin.kafka.cruisecontrol.monitor.sampling.KafkaSampleStore.createProducer(KafkaSampleStore.java:181)
	at com.linkedin.kafka.cruisecontrol.monitor.sampling.KafkaSampleStore.configure(KafkaSampleStore.java:161)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.getConfiguredInstance(KafkaCruiseControlConfig.java:1413)
	at com.linkedin.kafka.cruisecontrol.monitor.task.LoadMonitorTaskRunner.<init>(LoadMonitorTaskRunner.java:104)
	at com.linkedin.kafka.cruisecontrol.monitor.task.LoadMonitorTaskRunner.<init>(LoadMonitorTaskRunner.java:74)
	at com.linkedin.kafka.cruisecontrol.monitor.LoadMonitor.<init>(LoadMonitor.java:177)
	at com.linkedin.kafka.cruisecontrol.monitor.LoadMonitor.<init>(LoadMonitor.java:127)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControl.<init>(KafkaCruiseControl.java:124)
	at com.linkedin.kafka.cruisecontrol.async.AsyncKafkaCruiseControl.<init>(AsyncKafkaCruiseControl.java:75)
	at com.linkedin.kafka.cruisecontrol.KafkaCruiseControlMain.main(KafkaCruiseControlMain.java:71)
```

**Comment by linehrr** [comment]
@ninthnails 
I had the same issue before, but turned out it's missing config settings. 
in your case, it's likely to be the same since I see this line in your ST: 
```	
at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.getConfiguredInstance(KafkaCruiseControlConfig.java:1413)
```
make sure you based on the latest branch's config example, since they are adding more required configs over time.

**Comment by ninthnails** [comment]
I made sure to picked the config example from the 2.0.50 branch/tags. I only changed some paths to config files to be absolute. The rest is exactly the same. It seems thought he master branch differ from the 2.0.50 tag. For example:
```
#capacity.config.file=config/capacity.json
capacity.config.file=config/capacityJBOD.json
```
So 2.0.50 use the `capacityJBOD.json` file and master `capacity.json`. I'm still using the non JBOD format.

**Comment by linehrr** [comment]
this is where it throws.
```    
producerProps.setProperty(ProducerConfig.RECONNECT_BACKOFF_MS_CONFIG,
                              (String) config.get(KafkaCruiseControlConfig.RECONNECT_BACKOFF_MS_CONFIG));
```
and looking at `2.0.50` config, `#reconnect.backoff.ms=50` is commented out, just uncomment this line in your config and try.

**Comment by ninthnails** [comment]
@linehrr 
Finally got it working by de-commenting `#reconnect.backoff.ms=50` as you pointed out. Thanks mate!

**Comment by linehrr** [comment]
@ninthnails 
it is a bug, we shouldn't need to uncomment out this, according to the document, there should be a default value. I found where the pitfall was, the config loading path in the code base is a bit messy, I will do a PR and properly fix this, but at the moment, you will have to explicitly specify this in your config.

**Comment by kidkun** [comment]
@ninthnails @linehrr  I am terribly sorry that PR #735  does not fixes this NPE bug and published the flawed version 2.0.50/0.1.53, I have just merged the fix.
